src/lib.rs

```diff
@@ -263,18 +282,67 @@ impl Scope {
     }
 
     /// Push a new module definition, returning a mutable reference to it.
+    /// 
+    /// # Panics
+    /// 
+    /// Since a module's name must uniquely identify it within the scope in 
+    /// which it is defined, pushing a module whose name is already defined
+    /// in this scope will cause this function to panic.
+    /// 
+    /// In many cases, the [`get_or_add_module`] function is preferrable, as it 
+    /// will return the existing definition instead.
+    /// 
+    /// [`get_or_add_module`]: #method.get_or_add_module
     pub fn new_module(&mut self, name: &str) -> &mut Module {
         self.push_module(Module::new(name));
 
-        match *self.items.last_mut().unwrap() {
-            Item::Module(ref mut v) => v,
-            _ => unreachable!(),
+        &mut self.modules[name]
+    }
+
+    /// Returns a mutable reference to a module if it is exists in this scope. 
+    pub fn get_module_mut<Q: ?Sized>(&mut self, 
+                                     name: &Q) 
+                                     -> Option<&mut Module> 
+    where
+        Q: Hash + ordermap::Equivalent<String>,
+    {
+        self.modules.get_mut(name)
+    }
+
+    /// Returns a mutable reference to a module if it is exists in this scope. 
+    pub fn get_module<Q: ?Sized>(&self, name: &Q) -> Option<&Module>
+    where
+        Q: Hash + ordermap::Equivalent<String>,
+    {
+        self.modules.get(name)
+    }
+
+    /// Returns a mutable reference to a module, creating it if it does 
+    /// not exist.
+    pub fn get_or_add_module(&mut self, name: &str) -> &mut Module {
```

Comment:
Could this be named `get_or_new_module` in this PR?

I agree the `new` terminology is not ideal, but that is the convention currently used for all other fns. We can approach improving the naming scheme in another PR.

Reply:
Sure, changed it back in 3824853d2c6e39b53c7c29785689d361b0c7232f.